rhp/v4/validation.go

```diff
@@ -149,6 +151,38 @@ func (req *RPCRenewContractRequest) Validate(pk types.PublicKey, tip types.Chain
 	}
 }
 
+// Validate validates a renew contract request. Prices are not validated
+func (req *RPCRefreshContractRequest) Validate(pk types.PublicKey, expirationHeight uint64, maxCollateral types.Currency) error {
+	if err := req.Prices.Validate(pk); err != nil {
+		return fmt.Errorf("prices are invalid: %w", err)
+	}
+
+	// validate the request fields
+	switch {
+	case req.MinerFee.IsZero():
+		return errors.New("miner fee must be greater than 0")
+	case req.Basis == (types.ChainIndex{}):
+		return errors.New("basis must be set")
+	}
+
+	// validate the contract fields
+	hp := req.Prices
+	// calculate the minimum allowance required for the contract based on the
+	// host's locked collateral and the contract duration
+	minRenterAllowance := MinRenterAllowance(hp, expirationHeight-req.Prices.TipHeight, req.Refresh.Collateral)
+
+	switch {
+	case req.Refresh.Allowance.IsZero():
+		return errors.New("allowance is zero")
```

Comment:
We have two styles of error messages. Sometimes we state the value and indicate it's invalid, while other times we specify that 'value X must adhere to condition Y,' and explain that the condition was not met. I prefer the latter, where we would error out here with "allowance must be greater than zero". I'm going to get shamed for this `nit` but I don't care.